src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -33,26 +34,48 @@ public DeleteCommand(Index targetIndex) {
     public CommandResult executeUndoableCommand() throws CommandException {
 
         List<ReadOnlyPerson> lastShownList = model.getFilteredPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        for(Index targetIndex : targetIndex) {
+            if (targetIndex.getZeroBased() >= lastShownList.size()) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
         }
 
-        ReadOnlyPerson personToDelete = lastShownList.get(targetIndex.getZeroBased());
+        if (targetIndex.length > 1) {
+            for(int i=1; i<targetIndex.length; i++) {
+                if(targetIndex[i].getZeroBased() < targetIndex[i-1].getZeroBased()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_ORDER_PERSONS_INDEX);
+                }
+                else if (targetIndex[i].getZeroBased() == targetIndex[i-1].getZeroBased()) {
+                    throw new CommandException(Messages.MESSAGE_REPEATED_INDEXES);
+                }
+            }
+        }
 
-        try {
-            model.deletePerson(personToDelete);
-        } catch (PersonNotFoundException pnfe) {
-            assert false : "The target person cannot be missing";
+        ReadOnlyPerson personToDelete;
+
+        int i = 0;
+        StringBuilder deleteMessage = new StringBuilder();
+        for(Index targetIndex : targetIndex) {
+            int target = targetIndex.getZeroBased() - i;
```

Comment:
If you delete from the back to the front instead of deleting front to back, the 'i' can be removed.
Currently for index 1, 3 5, you call delete for index 1, 2, 3 in that order to account for the change in indexes after each deletion.
If you delete from the back to front, you call delete for index 5, 3, 1 without accounting for the change in indexes after each deletion